src/Conversion/ONNXToTOSA/Math/Elementwise.cpp

```diff
@@ -170,14 +197,41 @@ class ONNXLeakyReluOpLoweringToTOSA
 
 } // namespace
 
-void populateLoweringONNXElementwiseOpToTOSAPattern(ConversionTarget &target,
+static void populateLoweringONNXElementwiseBinaryTemplateOpToTOSAPattern(
     RewritePatternSet &patterns, TypeConverter &typeConverter,
     MLIRContext *ctx) {
-  patterns.insert<ONNXElementwiseUnaryOpLoweringToTOSA<ONNXNegOp>,
+  patterns.insert<
       ONNXBinaryElementwiseOpLoweringToTOSA<ONNXAddOp, mlir::tosa::AddOp>,
-      ONNXBinaryElementwiseOpLoweringToTOSA<ONNXSubOp, mlir::tosa::SubOp>,
-      ONNXFloorOpLoweringToTOSA, ONNXReluOpLoweringToTOSA,
-      ONNXLeakyReluOpLoweringToTOSA>(typeConverter, ctx);
+      ONNXBinaryElementwiseOpLoweringToTOSA<ONNXSubOp, mlir::tosa::SubOp>>(
+      typeConverter, ctx);
+}
+
+static void populateLoweringONNXElementwiseUnaryTemplateOpToTOSAPattern(
+    RewritePatternSet &patterns, TypeConverter &typeConverter,
+    MLIRContext *ctx) {
+  patterns.insert<
+      ONNXElementwiseUnaryOpLoweringToTOSA<ONNXNegOp, mlir::tosa::NegateOp>,
+      ONNXElementwiseUnaryOpLoweringToTOSA<ONNXCeilOp, mlir::tosa::CeilOp>,
```

Comment:
The Floor legalization has its own pattern because it only accepts float as input. However the onnx op also only accepts floats. Should I move the floor pattern in here or should I create a new template which checks for floats? WDYT @ljfitz ?

Reply:
I'd suggest extending `ONNXElementwiseUnaryOpLoweringToTOSA` with a template parameter to perform different type checking if appropriate so it can continue to share the rest of the rewriting.